pkg/controller/history/controller_history.go

```diff
@@ -332,10 +332,14 @@ func (rh *realHistory) AdoptControllerRevision(parent metav1.Object, parentKind
 }
 
 func (rh *realHistory) ReleaseControllerRevision(parent metav1.Object, revision *apps.ControllerRevision) (*apps.ControllerRevision, error) {
+	dataBytes, err := generatePatchBytesForDelete(parent.GetUID(), revision.UID)
+	if err != nil {
+		return nil, err
+	}
+
 	// Use strategic merge patch to add an owner reference indicating a controller ref
 	released, err := rh.client.AppsV1().ControllerRevisions(revision.GetNamespace()).Patch(context.TODO(), revision.GetName(),
-		types.StrategicMergePatchType,
-		[]byte(fmt.Sprintf(`{"metadata":{"ownerReferences":[{"$patch":"delete","uid":"%s"}],"uid":"%s"}}`, parent.GetUID(), revision.UID)), metav1.PatchOptions{})
```

Comment:
Another approach is probably to use `%q` while building the string, but even that can send out invalid json up the stack. I sort of prefer this way as it guarantees json and also adds a few tests.